source/common/upstream/health_discovery_service.cc

```diff
@@ -68,10 +169,90 @@ void HdsDelegate::onReceiveTrailingMetadata(Http::HeaderMapPtr&& metadata) {
 
 void HdsDelegate::onRemoteClose(Grpc::Status::GrpcStatus status, const std::string& message) {
   ENVOY_LOG(warn, "gRPC config stream closed: {}, {}", status, message);
-  response_timer_->disableTimer();
+  server_response_timer_->disableTimer();
   stream_ = nullptr;
   handleFailure();
 }
 
+HdsCluster::HdsCluster(Runtime::Loader& runtime, const envoy::api::v2::Cluster& cluster,
+                       const envoy::api::v2::core::BindConfig& bind_config, Stats::Store& stats,
+                       Ssl::ContextManager& ssl_context_manager,
+                       Secret::SecretManager& secret_manager, bool added_via_api,
+                       HdsInfoFactory& info_factory)
+    : runtime_(runtime), cluster_(cluster), bind_config_(bind_config), stats_(stats),
+      ssl_context_manager_(ssl_context_manager), secret_manager_(secret_manager),
+      added_via_api_(added_via_api), initial_hosts_(new HostVector()), info_factory_(info_factory) {
+  ENVOY_LOG(debug, "Creating an HdsCluster");
+  priority_set_.getOrCreateHostSet(0);
+
+  ClusterInfoConstSharedPtr info_ptr;
+  info_ptr =
+      info_factory_.createHdsClusterInfo(runtime_, cluster_, bind_config_, stats_,
+                                         ssl_context_manager_, secret_manager_, added_via_api_);
+  info_.swap(info_ptr);
```

Comment:
This seems a bit indirect. Can't we just write something like `info_ = info_factory_.createHds...`?